server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -261,11 +261,17 @@ private void closePrevotingAndElectionScheduler() {
     private void updateMaxTermSeen(final long term) {
         final long updatedMaxTermSeen = maxTermSeen.updateAndGet(oldMaxTerm -> Math.max(oldMaxTerm, term));
         synchronized (mutex) {
-            if (mode == Mode.LEADER && publicationInProgress() == false && updatedMaxTermSeen > getCurrentTerm()) {
-                // Bump our term. However if there is a publication in flight then doing so would cancel the publication, so don't do that
-                // since we check whether a term bump is needed at the end of the publication too.
-                ensureTermAtLeast(getLocalNode(), updatedMaxTermSeen);
-                startElection();
+            long currentTerm = getCurrentTerm();
+            if (mode == Mode.LEADER && updatedMaxTermSeen > currentTerm) {
+                if (publicationInProgress() == false) {
+                    // Bump our term. However if there is a publication in flight then doing so would cancel the publication, so don't do that
```

Comment:
maybe move that comment before the `if (publicationInProgress() == false)` check